openmmml/models/macepotential.py

```diff
@@ -0,0 +1,222 @@
+
+                self.model = jit.compile(self.model)
+                
+                # setup input
+                N = len(atomic_numbers)
+                self.ptr = torch.nn.Parameter(torch.tensor([0,N], dtype=torch.long), requires_grad=False)
+                self.batch = torch.nn.Parameter(torch.zeros(N, dtype=torch.long), requires_grad=False)
+                
+                # one hot encoding of atomic number
+                self.node_attrs = torch.nn.Parameter(to_one_hot(
+                        torch.tensor(atomic_numbers_to_indices(atomic_numbers, z_table=self.z_table), dtype=torch.long).unsqueeze(-1),
+                        num_classes=len(self.z_table),
+                    ), requires_grad=False)
+
+                if periodic:
+                    self.pbc = torch.nn.Parameter(torch.tensor([True, True, True]), requires_grad=False)
+                else:
+                    self.pbc = torch.nn.Parameter(torch.tensor([False, False, False]), requires_grad=False)
+
+
+                if indices is None:
+                    self.indices = None
+                else:
+                    self.indices = torch.tensor(indices, dtype=torch.int64)
+
+                if interaction_energy is True:
+                    self.return_energy_type = "interaction_energy"
+                else:
+                    self.return_energy_type = "energy"
+
+            def forward(self, positions, boxvectors: Optional[torch.Tensor] = None):
+                
+                # setup positions
+                positions = positions.to(self.default_dtype)
+                if self.indices is not None:
+                    positions = positions[self.indices]
+
+                positions = positions*self.nm_to_distance
+
+                if boxvectors is not None:
+                    cell = boxvectors.to(self.default_dtype) * self.nm_to_distance
+                    pbc = True
+                else:
+                    cell = torch.eye(3, device=positions.device)
+                    pbc = False
+  
+                mapping, shifts_idx = simple_nl(positions, cell, pbc, self.model.r_max)
+                
+                edge_index = torch.stack((mapping[0], mapping[1]))
+
+                shifts = torch.mm(shifts_idx, cell)
+
+                # create input dict
+                input_dict = { "ptr" : self.ptr, 
+                              "node_attrs": self.node_attrs, 
+                              "batch": self.batch, 
+                              "pbc": self.pbc,
+                              "cell": cell,
+                              "positions": positions,
+                              "edge_index": edge_index,
+                              "unit_shifts": shifts_idx,
+                              "shifts": shifts}
+                
+                # predict
+                out = self.model(input_dict,compute_force=False)
+
+                energy = out[self.return_energy_type]
+                if energy is None:
```

Comment:
How can you get energy `None`? The simulation should fail rather than silently assume it is 0.